controllers/cache/scale_up.go

```diff
@@ -0,0 +1,72 @@
+package cache
+
+import (
+	"fmt"
+	redisCli "github.com/spotahome/redis-operator/api/redisfailover/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime"
+
+	//appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+)
+
+// UpScale reconcile will upscale Redis sentinel cluster if replicas upscale.
+// It does:
+// - check replicas
+// - update RedisFailovers CR replicas
+func (redis *RedisReconciler) ScaleUpCache() error {
+
+	crdClient := redis.DClient.WithResource(redisFailoversGVR).WithNamespace(redis.Namespace)
```

Comment:
It seems there are some similar codes in the `ScaleUpCache` and `ScaleDownCache` methods, is it possible to reorganize code to reduce duplicated parts?